katdal/schemas/__init__.py

```diff
@@ -0,0 +1,58 @@
+"""Makes packaged XSD schemas available as validators."""
+
+import pkg_resources
+
+
+has_lxml = False
+validators = {}
+
+try:
+    from lxml import etree
+    has_lxml = True
+except ImportError:
+    pass
+
+
+class ValidatorWithLog(object):
+    def __init__(self, validator):
+        self.validator = validator
+
+    def validate(self, xml_string):
+        """Validates a supplied XML string against the instantiated validator.
+
+        Parameters
+        ---------
+        xml_string : str
+            String representation of the XML to be turned into a document
+            and validated.
+
+        Raises
+        ------
+        etree.DocumentInvalid
+            if `xml_string` does not validate against the XSD schema
+        ValueError
+            if `xml_string` cannot be parsed into a valid XML document
+        """
+        try:
+            xml_doc = etree.fromstring(bytes(bytearray(xml_string, encoding='utf-8')))
```

Comment:
Out of curiosity, why `bytes` and `bytesarray`?

Reply:
Python2.7 doesn't allow encoding types in bytes, but does in bytearray. Kludge to keep 2/3 compatible.

Reply:
There should be something in `future.utils` to force things one way or another.

But also, why encode at all? lxml.etree.fromstring seems perfectly happy to take Unicode.